### PR TITLE
Execute _start func before benches.

### PR DESCRIPTION
**Pull request by mathetake**
Previously, we didn't execute _start function which is in charge of
in-Wasm memory state initialization.

That's why we faced the memory related error with `-gc=conservative` 
which is the default validator.

Also this commit revives the file system example which was temporarily removed.

### CODE REVIEW

File: examples/host_func_test.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func Test_hostFunc(t *testing.T) {
 
 	// Host-side implementation of get_random_string on Wasm import.
 	getRandomString := func(ctx *wasm.HostFunctionCallContext, retBufPtr uint32, retBufSize uint32) {
-		const bufferSize = 10
+		const bufferSize = 10000
```
Comment (mathetake) [inline examples/host_func_test.go:28]: This makes it necessary for Wasm program to grow memory space, and the growing memory fails when  we don't execute _start function beforehand. That's it!

File: examples/file_system_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package examples
```
Comment (mathetake) [inline examples/file_system_test.go:1]: revived the file_system test!

File: bench/bench_test.go [hunk 3]
```diff
@@ -15,41 +15,50 @@ import (
...
+	mod, err := wasm.DecodeModule((buf))
+	if err != nil {
+		panic(err)
+	}
+	err = store.Instantiate(mod, "test")
+	if err != nil {
+		panic(err)
+	}
+
+	// We assume that TinyGo binary expose "_start" symbol
```
Comment (codefromthecrypt) [inline bench/bench_test.go:44]: is there an api to check existence of a function? if so, we could still do this, and work if we switch the wasm implementation.

File: examples/file_system_test.go (hunk 2 above)
Comment (codefromthecrypt) [reply examples/file_system_test.go:1]: welcome back

File: examples/host_func_test.go (hunk 1 above)
Comment (codefromthecrypt) [reply examples/host_func_test.go:28]: ok so this fails nicer in missing `_start` in other words, right?

```suggestion
		const bufferSize = 10000 // force memory space grow to ensure eager failures on missing setup 
```

File: examples/host_func_test.go [hunk 4]
```diff
@@ -54,6 +54,13 @@ func Test_hostFunc(t *testing.T) {
 	err = store.Instantiate(mod, "test")
 	require.NoError(t, err)
 
+	// We assume that TinyGo binary expose "_start" symbol
+	// to initialize the memory state.
+	// Meaning that TinyGo binary is "WASI command":
+	// https://github.com/WebAssembly/WASI/blob/main/design/application-abi.md
+	_, _, err = store.CallFunction("test", "_start")
```
Comment (codefromthecrypt) [inline examples/host_func_test.go:61]: same suggestion as above, if we can detect presence of a function

**Review by codefromthecrypt — APPROVED** [review]
Looks good only one suggestion, but not sure it is possible

File: bench/bench_test.go (hunk 3 above)
Comment (mathetake) [reply bench/bench_test.go:44]: we don't have such API but the CallFunction returns error in these case. I would create specific error for these cases `ErrFunctionNotFound`, but leave it for other PRs!

Comment (mathetake) [reply bench/bench_test.go:44]: https://github.com/tetratelabs/wazero/blob/main/wasm/store.go#L162-L170

Comment (codefromthecrypt) [reply bench/bench_test.go:44]: I will raise a PR for `HasFunction` or similar!

**Pull request merged**